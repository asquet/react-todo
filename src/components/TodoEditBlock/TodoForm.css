.TodoForm {
  text-align: center;
  margin-left: 10px;
  margin-right: 10px;
}

.TodoForm input.TodoForm-textInput {
  width: 400px;
}

@media (max-width: 767px) {
  .TodoForm input.TodoForm-textInput {
    box-sizing: border-box;
    margin: auto;
    width: 100%;
  }
}
