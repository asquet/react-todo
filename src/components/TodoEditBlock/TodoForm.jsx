import React from 'react';

function TodoForm({ onSave, onTextChange, onCancel, formText }) {
  return (
    <form
      onSubmit={(ev) => {
        ev.preventDefault();
        onSave(formText);
      }}
      className=""
    >
      <div className="form-group">
        <label htmlFor="todoTextInput">Todo text</label>
        <input
          className="form-control"
          type="text"
          id="todoTextInput"
          value={formText}
          onChange={(ev) => {
            ev.preventDefault();
            onTextChange(ev.target.value);
          }}
        />
      </div>
      <div>
        <button type="submit" className="btn btn-primary">Save</button>
        <button
          onClick={(ev) => {
            ev.preventDefault();
            onTextChange('');
            onCancel();
          }}
          className="btn btn-default"
        >
          Cancel
        </button>
      </div>
    </form>
  );
}

TodoForm.propTypes = {
  onCancel: React.PropTypes.func.isRequired,
  onSave: React.PropTypes.func.isRequired,
  onTextChange: React.PropTypes.func.isRequired,
  formText: React.PropTypes.string.isRequired
};

export default TodoForm;
