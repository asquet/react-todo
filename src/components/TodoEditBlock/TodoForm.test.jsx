import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import TodoForm from './TodoForm';

const f = () => {
};

function renderCmp(foreignConfig) {
  const config = Object.assign(
    {
      onCancel: f,
      onSave: f,
      onTextChange: f,
      formText: ''
    }, foreignConfig
  );

  return (<TodoForm
    onCancel={config.onCancel}
    onSave={config.onSave}
    onTextChange={config.onTextChange}
    formText={config.formText}
  />);
}

it(
  'renders without crashing', () => {
    const focusListener = jest.fn();

    const component = renderer.create(
      renderCmp(), {
        createNodeMock: (element) => {
          if (element.type === 'input') {
            return {
              focus: focusListener
            };
          }
          return null;
        }
      }
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(focusListener.mock.calls.length).toBe(1);
  }
);

it(
  'responds to typing', () => {
    let rendered;
    const onTextChange = jest.fn(
      (text) => {
        rendered.setProps(
          {
            formText: text
          }
        );
      }
    );

    rendered = shallow(renderCmp({ onTextChange, formText: '' }));

    rendered.find('input').simulate(
      'change', {
        target: { value: '1' },
        preventDefault: () => {
        }
      }
    );
    expect(onTextChange.mock.calls[0][0]).toBe('1');
    expect(onTextChange.mock.calls.length).toBe(1);
    expect(rendered.find('input').props().value).toEqual('1');
  }
);

it(
  'responds to for submit', () => {
    const onSave = jest.fn();
    const onTextChange = jest.fn();

    const rendered = mount(renderCmp({ onSave, onTextChange, formText: 'text 1' }));

    rendered.find('[type="submit"]').get(0).click(
      {
        preventDefault: () => {
        }
      }
    );

    expect(onSave.mock.calls.length).toBe(1);
    expect(onSave.mock.calls[0][0]).toBe('text 1');

    expect(onTextChange.mock.calls.length).toBe(1);
  }
);


it(
  'responds to cancel btn click', () => {
    const onCancel = jest.fn();
    const onTextChange = jest.fn();

    const rendered = shallow(renderCmp({ onCancel, onTextChange, formText: 'text 1' }));

    rendered.find('button.btn-default').simulate(
      'click', {
        preventDefault: () => {
        }
      }
    );

    expect(onCancel.mock.calls.length).toBe(1);

    expect(onTextChange.mock.calls.length).toBe(1);
    expect(onTextChange.mock.calls[0][0]).toBe('');
  }
);
