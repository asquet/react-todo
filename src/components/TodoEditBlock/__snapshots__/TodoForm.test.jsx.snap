exports[`test renders without crashing 1`] = `
<form
  className="TodoForm"
  onSubmit={[Function]}>
  <div
    className="form-group">
    <label
      htmlFor="todoTextInput">
      Todo text
    </label>
     
    <input
      className="form-control TodoForm-textInput"
      id="todoTextInput"
      onChange={[Function]}
      type="text"
      value="" />
  </div>
   
  <button
    className="btn btn-primary"
    type="submit">
    Save
  </button>
   
  <button
    className="btn btn-default"
    onClick={[Function]}>
    Cancel
  </button>
</form>
`;
